foxglove_bridge_base/include/foxglove_bridge/common.hpp

```diff
@@ -74,26 +75,28 @@ struct ClientMessage {
   uint64_t logTime;
   uint64_t publishTime;
   uint32_t sequence;
-  const ClientAdvertisement& advertisement;
+  ClientAdvertisement advertisement;
   size_t dataLength;
-  const uint8_t* data;
+  std::vector<uint8_t> data;
 
   ClientMessage(uint64_t logTime, uint64_t publishTime, uint32_t sequence,
-                const ClientAdvertisement& advertisement, size_t dataLength, const uint8_t* data)
+                const ClientAdvertisement& advertisement, size_t dataLength, const uint8_t* rawData)
       : logTime(logTime)
       , publishTime(publishTime)
       , sequence(sequence)
       , advertisement(advertisement)
       , dataLength(dataLength)
-      , data(data) {}
+      , data(dataLength) {
+    std::memcpy(data.data(), rawData, dataLength);
```

Comment:
Is the additional copy worth it? We should test on a low-power platform like a Jetson Nano.

Reply:
We have to do a copy now since we call the handler in a different thread and the pointer to the original data is likely not valid anymore.